v2/car.go

```diff
@@ -17,16 +22,17 @@ var (
 		0x02, // uint(2)
 	}
 	// The size of the CAR v2 prefix in 11 bytes, (i.e. 11).
-	PrefixBytesSize = len(PrefixBytes)
+	PrefixBytesSize = uint64(len(PrefixBytes))
 	// Reserved 128 bits space to capture future characteristics of CAR v2 such as order, duplication, etc.
 	EmptyCharacteristics = new(Characteristics)
 )
 
 type (
 	// Header represents the CAR v2 header/pragma.
 	Header struct {
+		io.WriterTo
 		// 128-bit characteristics of this CAR v2 file, such as order, deduplication, etc. Reserved for future use.
-		Characteristics Characteristics
+		Characteristics *Characteristics
```

Comment:
why would this be a pointer? if you want to be able to cast a memory region into a car header, you'd expect it to be the direct bytes